pkg/startup/startup.go

```diff
@@ -168,11 +189,13 @@ func Run() {
 		terminate()
 	}
 
+	kubeadmConfig, _ := queryKubeadmConfigMap(clientset)
```

Comment:
We definitely need to handle the error here. What happens if we try to query the configmap but hit an error connecting to or authorizing with the api server? I think we need to fail in that scenario.

The only errors we don't want to fail is:
- If we're not running on k8s
- If we get a "Does not exist" error.

Reply:
Check the implementation of `queryKubeadmConfigMap()`, the error handling for the API call is in there.

Reply:
Also, if we move this within InClusterConfig block above, then we won't need the "If clientset == nil" check within the function. It will let us group all of the Kuberentes-specific logic in one place.

Reply:
> Check the implementation of queryKubeadmConfigMap(), the error handling for the API call is in there.

But, it looks like the function returns an error but it's never checked anywhere, is that right?